QuizApp/QuizApp/UI/Components/CustomSegmentedControl.swift

```diff
@@ -0,0 +1,90 @@
+import UIKit
+
+protocol CustomSegmentedControlDelegate: AnyObject {
+    func segmentTapped(view: SegmentView)
+}
+
+class CustomSegmentedControl: UIView {
+
+    private var scrollView: UIScrollView!
+    private var stackView: UIStackView!
+
+    weak var delegate: CustomSegmentedControlDelegate!
+
+    init() {
+        super.init(frame: .zero)
+
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func loadData(data: [(String, UIColor)]) {
```

Comment:
Avoid defining this kind of data structure. Instead, create a new struct, like:
```
struct CustomSegmentedControlModel {
   
    let id: Any
    let title: String
    let color: UIColor
    let isActive: Bool

}
```

Also, rename to `func set(data: [])`